src/mapogcsld.cpp

```diff
@@ -147,6 +148,58 @@ int msSLDApplySLDURL(mapObj *map, const char *szURL, int iLayer,
 #endif
 }
 
+/************************************************************************/
+/*                             msSLDApplyFromFile                       */
+/*                                                                      */
+/*      Apply SLD from a file on disk to the layerObj                   */
+/************************************************************************/
+int msSLDApplyFromFile(mapObj *map, layerObj *layer, char *filename) {
+
+  FILE *fp = NULL;
+
+  int nStatus = MS_FAILURE;
+  const char *realpath;
+
+  /* open SLD file */
+  char szPath[MS_MAXPATHLEN];
+  char *pszSLDbuf = NULL;
+  realpath = msBuildPath(szPath, map->mappath, filename);
+
+  if ((fp = fopen(realpath, "rb")) != NULL) {
+    int nBufsize = 0;
+    fseek(fp, 0, SEEK_END);
+    nBufsize = ftell(fp);
+    if (nBufsize > 0) {
+      rewind(fp);
+      pszSLDbuf = (char *)malloc((nBufsize + 1) * sizeof(char));
```

Comment:
it would prudent to check pszSLDbuf for null

Reply:
Added a check to cover both NULL and 0 (which was hinted as a possible error) using the following code. If this check makes sense I can add it to the original function I copied it from. 

```c
      if (!pszSLDbuf) {
        msSetError(MS_MEMERR, "Failed to read SLD file.",
                   "msSLDApplyFromFile()");
        nStatus = MS_FAILURE;
      
```

Reply:
> to cover both NULL and 0 

not sure to understand what 0 means here?

actually a very edgy case would be nBufSize == INT_MAX, in which case nBufSize + 1 would overflow...

Reply:
Checking just for NULL gets rid of the warning in VS
![image](https://github.com/MapServer/MapServer/assets/490840/d0d811bf-9a02-4022-9d47-787e7f64cf7d)

This code is used in a couple of other places in MapServer.